### PR TITLE
network: add support for br content encoding

### PR DESCRIPTION
**Pull request by thc202**
Support encoding/decoding on Linux x64, macOS, and Windows x64.

Part of zaproxy/zaproxy#2198.

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/ContentEncodingsHandler.java [hunk 1]
```diff
@@ -40,6 +40,8 @@ public void handle(HttpHeader header, HttpBody body) {
             encodings = List.of(HttpEncodingDeflate.getSingleton());
         } else if (encoding.contains(HttpHeader.GZIP)) {
             encodings = List.of(HttpEncodingGzip.getSingleton());
+        } else if (HttpEncodingBrotli.isAvailable() && encoding.contains("br")) {
```
Comment (ricekot) [inline addOns/network/src/main/java/org/zaproxy/addon/network/internal/ContentEncodingsHandler.java:43]: Reported in IRC, one response may be encoded multiple times (ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding), but that can probably be tackled in another PR.

File: addOns/network/src/main/javahelp/help/contents/options/localservers.html [hunk 2]
```diff
@@ -61,7 +61,13 @@ <H4>Behind NAT</H4>
 	<H4>Remove Accept-Encoding Request Header</H4>
 	Allows the proxy to remove the "Accept-Encoding" request-header field, so no (unsupported) encoding transformations are done to the response.<br/>
 	This option should be always enabled unless when testing the encoding transformations.<br/>
-	The messages encoded with unsupported encodings will not be correctly scanned (either by passive and active scanners).
+	The messages encoded with unsupported encodings will not be correctly scanned (either by passive and active scanners).<br/>
```
Comment (ricekot) [inline addOns/network/src/main/javahelp/help/contents/options/localservers.html:64]:
```suggestion
	The messages encoded with unsupported encodings will not be correctly scanned (either by passive or active scanners).<br/>
```

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/HttpEncodingBrotli.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import com.aayushatharva.brotli4j.decoder.DecoderJNI;
+import com.aayushatharva.brotli4j.decoder.DirectDecompress;
+import com.aayushatharva.brotli4j.encoder.Encoder;
+import java.io.IOException;
+import org.zaproxy.zap.network.HttpEncoding;
+
+/** The {@link HttpEncoding} for the {@code br} coding. */
+class HttpEncodingBrotli implements HttpEncoding {
+
+    private static final HttpEncodingBrotli SINGLETON = new HttpEncodingBrotli();
```
Comment (ricekot) [inline addOns/network/src/main/java/org/zaproxy/addon/network/internal/HttpEncodingBrotli.java:33]: Is this thread-safe?

**Review by ricekot — COMMENTED** [review]
Working as expected :), tested with the following request:
```
GET https://www.zaproxy.org/ HTTP/2
accept-encoding: br
content-length: 0

```
Have a few other minor comments.

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/HttpEncodingBrotli.java (hunk 3 above)
Comment (thc202) [reply addOns/network/src/main/java/org/zaproxy/addon/network/internal/HttpEncodingBrotli.java:33]: You mean the initialisation? The implementation is, there's no state being shared.

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/ContentEncodingsHandler.java (hunk 1 above)
Comment (thc202) [reply addOns/network/src/main/java/org/zaproxy/addon/network/internal/ContentEncodingsHandler.java:43]: Yeah, will fix in other PR.

File: addOns/network/src/main/javahelp/help/contents/options/localservers.html (hunk 2 above)
Comment (thc202) [reply addOns/network/src/main/javahelp/help/contents/options/localservers.html]: Changed.

**Review by ricekot — APPROVED** [review]
Thank you!

**Review by psiinon — APPROVED** [review]
Works on my macBook M2!
